### PR TITLE
Replace I/O Sprintf with string templates

### PR DESCRIPTION
**Pull request by BuddhiWathsala**
## Purpose
$subject

## Approach
For
- strings BBE
- gRPC integration tests

### CODE REVIEW

File: examples/strings/strings.bal [hunk 1]
```diff
@@ -64,18 +64,20 @@ public function main() {
     boolean contains = statement.includes("Town", 7);
     io:println("Contains: ", contains);
 
-    // Format a `string` according to the given format arguments.
+    // Format a `string` according to the given format arguments
+    // using string templates.
     string name = "Sam";
-    int marks = 90;
+    int mrk = 90;
```
Comment (praveennadarajah) [inline examples/strings/strings.bal:70]:
```suggestion
    int marks = 90;
```

File: examples/strings/strings.bal [hunk 2]
```diff
@@ -64,18 +64,20 @@ public function main() {
...
     string[] subjects = ["English", "Science"];
-    float average = 71.5;
-    string s8 = io:sprintf("%s scored %d for %s and has an average of %.2f.",
-     name, marks, subjects[0], average);
-    io:println("Sprintf: ", s8);
+    float avg = 71.5;
```
Comment (praveennadarajah) [inline examples/strings/strings.bal:72]:
```suggestion
    float averge = 71.5;
```

File: examples/strings/strings.bal [hunk 3]
```diff
@@ -64,18 +64,20 @@ public function main() {
...
+    string s8 = string `${name} scored ${mrk} for ${subjects[0]} and has an
+                  average of ${avg}`;
```
Comment (praveennadarajah) [inline examples/strings/strings.bal:74]: Shall we adjust here in order to get 80 characters per line?

File: examples/strings/strings.bal (hunk 1 above)
Comment (BuddhiWathsala) [reply examples/strings/strings.bal]: Fixed in here https://github.com/ballerina-platform/ballerina-distribution/pull/1442/commits/cd8c2d975eb65d4905cb6169f946b82be90e5204

File: examples/strings/strings.bal (hunk 2 above)
Comment (BuddhiWathsala) [reply examples/strings/strings.bal]: Fixed in here https://github.com/ballerina-platform/ballerina-distribution/pull/1442/commits/cd8c2d975eb65d4905cb6169f946b82be90e5204

File: examples/strings/strings.bal (hunk 3 above)
Comment (BuddhiWathsala) [reply examples/strings/strings.bal]: It is within 80 characters.

**Review by daneshk — APPROVED** [review]

**Review by praveennadarajah — APPROVED** [review]

**Pull request merged**